### DISCUSSION TITLE
Trouble with Hardcoded Paths

### DISCUSSION THREAD

**Discussion by will-leeson** [soot-oss/SootUp#551] [Q&A] [answered]
Hi, I'm new to SootUp (and Soot in general). I'm trying to get started by testing out building Call graphs and CFGs. I copied the code from  the [CallGraph Example](https://github.com/soot-oss/SootUp/blob/develop/sootup.examples/src/test/java/sootup/examples/callgraph/CallgraphExample.java) and changed the path to a hardcoded location [/path/to/folder] and [folder] includes the files A.java, B.java, C.java, and D.java from the [Call Graph Resources](https://github.com/soot-oss/SootUp/tree/develop/sootup.examples/src/test/resources/Callgraph/source) directory.

I can compile this fine, but when I execute it I get the following error:

```
[main] INFO sootup.core.typehierarchy.ViewTypeHierarchy - Type hierarchy scan took 116.52449 ms
Exception in thread "main" sootup.core.frontend.ResolveException: Could not find A in hierarchy. ./file-does-not-exist No position info
	at sootup.core.typehierarchy.ViewTypeHierarchy.subclassesOf(ViewTypeHierarchy.java:87)
	at com.lesslab.ProgramGraph.App.main(App.java:72)
```

Is there a problem with changing the file? I am missing something?

**Comment by kadirayk** [comment, marked as answer]
Hi, are you using `JavaClassPathAnalysisInputLocation`? If so you should compile them into class files.

**Reply by will-leeson** [reply]
Java**Class**PathAnalysisInputLocation. I need to get better at reading. Thank you. This did fix it. 
I switched to JavaSourcePathAnalysisInputLocation and was able to do it without class files. Thank you
